Add example to using webhooks and email notifications on Databricks Workflow

tatiana opened the pull request:
As part of `extra_job_params`, we already supported `email_notifications` and `webhooks`.

Closes: #121

Example:
```
   task_group = DatabricksWorkflowTaskGroup(
        group_id="test_workflow",
        databricks_conn_id=DATABRICKS_CONN_ID,
        job_clusters=job_cluster_spec,
        notebook_params=[],
        notebook_packages=[
            {
                "pypi": {
                    "package": "simplejson==3.18.0",  # Pin specification version of a package like this.
                    "repo": "https://pypi.org/simple",  # You can specify your required Pypi index here.
                }
            },
        ],
        extra_job_params={
            "email_notifications": {
                "on_start": [DATABRICKS_NOTIFICATION_EMAIL],
            },
            "webhook_notifications": {
                "on_start": [{"id": DATABRICKS_DESTINATION_ID}],
            },
        },
    )
```

pankajkoti reviewed (commented):
The ticket #121  also suggests adding support for `email_notifications` which is another param as can be seen in the payload screenshot. Shouldn't we also allow users to configure this along with the `webhook_notifcations` parameter?

<img width="507" alt="Screenshot 2023-03-02 at 8 50 41 PM" src="https://user-images.githubusercontent.com/10206082/222471225-e44dee12-e31b-47ca-ba33-f18d1a2bd311.png">

dimberman reviewed (dismissed):
Ahh good catch @pankajkoti. Yes let's add emails as well.

tatiana commented:
@dimberman @pankajkoti I realised that email and webhooks were already supported by using `extra_job_params`, which I introduced in #153 . I simplified the code, updated the example DAG and saw things working locally.

pankajkoti commented on cosmos/providers/databricks/workflow.py:
```diff
@@ -112,8 +112,8 @@ def create_workflow_json(self) -> dict[str, object]:
         }
-        full_json = merge(full_json, self.extra_job_params)
-        return full_json
+        merged_json = merge(default_json, self.extra_job_params)
```
Looking at the constructor `__init__ ()`, `self.extra_job_params` could be `None` if it is not passed. We should initialise it to empty dictionary `{}` if it is `None`. Or else merge would give a `TypeError` here.

tatiana replied:
@pankajkoti the class `DatabricksWorkflowTaskGroup` was already handling that - but I'll add it to the Operator so we're on the safe side.

pankajkoti commented on cosmos/providers/databricks/workflow.py:
```diff
@@ -131,7 +131,9 @@ def execute(self, context: Context) -> Any:
         if not isinstance(self.task_group, DatabricksWorkflowTaskGroup):
-            raise AirflowException("Task group must be a DatabricksWorkflowTaskGroup")
+            raise AirflowException(
+                "Task group must be a Databricthat'sksWorkflowTaskGroup"
```
```suggestion
                "Task group must be a DatabricksWorkflowTaskGroup"
```

pankajkoti reviewed (approved):
This looks a lot cleaner and easier to support API params. Amazing! ❤️
